test failure with Go 1.25

### Description

```
time="2025-09-08T08:45:42+02:00" level=warning msg="error parsing level \"\": not a valid logrus Level: \"\", using \"info\"\t"
time="2025-09-08T08:45:42.632995308+02:00" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.25.0
time="2025-09-08T08:45:42.633009875+02:00" level=info msg="redis not configured" go.version=go1.25.0
time="2025-09-08T08:45:42.633083485+02:00" level=info msg="Starting upload purge in 59m0s" go.version=go1.25.0
time="2025-09-08T08:45:42.633230834+02:00" level=info msg="listening on [::]:5000" go.version=go1.25.0
time="2025-09-08T08:45:45.637333174+02:00" level=info msg="stopping server gracefully. Draining connections for 10s" go.version=go1.25.0
--- FAIL: TestGracefulShutdown (3.11s)
    registry_test.go:119: unexpected EOF
time="2025-09-08T08:45:45.78424222+02:00" level=warning msg="error parsing level \"\": not a valid logrus Level: \"\", using \"info\"\t"
time="2025-09-08T08:45:45.784712211+02:00" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.25.0
time="2025-09-08T08:45:45.784726898+02:00" level=info msg="redis not configured" go.version=go1.25.0
time="2025-09-08T08:45:45.784798414+02:00" level=info msg="Starting upload purge in 58m0s" go.version=go1.25.0
time="2025-09-08T08:45:45.78486476+02:00" level=info msg="restricting TLS version to tls1.2 or higher" go.version=go1.25.0
time="2025-09-08T08:45:45.784889206+02:00" level=info msg="restricting TLS cipher suites to: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256" go.version=go1.25.0
time="2025-09-08T08:45:45.785066693+02:00" level=info msg="listening on [::]:5001, tls" go.version=go1.25.0
time="2025-09-08T08:45:48.789102345+02:00" level=info msg="response completed" go.version=go1.25.0 http.request.host=127.0.0.1 http.request.id=01992812-dc74-76e8-bb2a-fe68a152a726 http.request.method=GET http.request.remoteaddr="127.0.0.1:54986" http.request.uri=/v2/ http.request.useragent= http.response.contenttype=application/json http.response.duration="647.747µs" http.response.status=200 http.response.written=2
127.0.0.1 - - [08/Sep/2025:08:45:48 +0200] "GET /v2/ HTTP/1.1" 200 2 "" ""
time="2025-09-08T08:45:48.789200711+02:00" level=info msg="stopping server gracefully. Draining connections for 10s" go.version=go1.25.0
time="2025-09-08T08:45:48.918574567+02:00" level=warning msg="error parsing level \"\": not a valid logrus Level: \"\", using \"info\"\t"
time="2025-09-08T08:45:48.919860384+02:00" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.25.0
time="2025-09-08T08:45:48.91991079+02:00" level=info msg="redis not configured" go.version=go1.25.0
time="2025-09-08T08:45:48.919934735+02:00" level=info msg="Starting upload purge in 15m0s" go.version=go1.25.0
time="2025-09-08T08:45:48.920148661+02:00" level=info msg="restricting TLS version to tls1.2 or higher" go.version=go1.25.0
time="2025-09-08T08:45:50.787569669+02:00" level=error msg="traces export: Post \"https://localhost:4318/v1/traces\": dial tcp [::1]:4318: connect: connection refused" go.version=go1.25.0
FAIL
FAIL	github.com/distribution/distribution/v3/registry	14.400s
```

### Reproduce

`go test ./registry/` with Go 1.25.0.
The same test passes with Go 1.24.5

### Expected behavior

tests pass

### registry version

3.0.0

### Additional Info

_No response_

--- a/registry/registry_test.go
+++ b/registry/registry_test.go
@@ -80,9 +80,8 @@ func TestGracefulShutdown(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-
 	// run registry server
-	var errchan chan error
+	errchan := make(chan error, 1)
 	go func() {
 		errchan <- registry.ListenAndServe()
 	}()
@@ -95,32 +94,39 @@ func TestGracefulShutdown(t *testing.T) {
 	// Wait for some unknown random time for server to start listening
 	time.Sleep(3 * time.Second)
 
-	// send incomplete request
+	//Establish connection
 	conn, err := net.Dial("tcp", "localhost:5000")
 	if err != nil {
 		t.Fatal(err)
 	}
-	fmt.Fprintf(conn, "GET /v2/ ")
 
+	// Send a complete, valid HTTP/1.1 request
+	_, err = fmt.Fprintf(conn, "GET /v2/ HTTP/1.1\r\nHost: localhost:5000\r\n\r\n")
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Give server time to receive and start processing the request
+	time.Sleep(100 * time.Millisecond)
 	// send stop signal
 	registry.quit <- os.Interrupt
+	// wait for shutdown to begin
 	time.Sleep(100 * time.Millisecond)
 
-	// try connecting again. it shouldn't
-	_, err = net.Dial("tcp", "localhost:5000")
+	// try connecting again - it should be rejected during graceful shutdown
+	conn2, err := net.Dial("tcp", "localhost:5000")
 	if err == nil {
+		conn2.Close()
 		t.Fatal("Managed to connect after stopping.")
 	}
 
-	// make sure earlier request is not disconnected and response can be received
-	fmt.Fprintf(conn, "HTTP/1.1\r\nHost: 127.0.0.1\r\n\r\n")
+	// Verify the original connection's request completes successfully
 	resp, err := http.ReadResponse(bufio.NewReader(conn), nil)
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer resp.Body.Close()
-	if resp.Status != "200 OK" {
-		t.Error("response status is not 200 OK: ", resp.Status)
+	if resp.StatusCode != 200 {
+		t.Errorf("Expected status 200 OK, got: %s ", resp.Status)
 	}
 	if body, err := io.ReadAll(resp.Body); err != nil || string(body) != "{}" {
 		t.Error("Body is not {}; ", string(body))

